homie/examples/lifecycle.rs

```diff
@@ -0,0 +1,49 @@
+use futures::FutureExt;
+use homie::HomieDevice;
+use rumqttc::MqttOptions;
+use std::error::Error;
+use std::time::Duration;
+use tokio::task::{self, JoinHandle};
+use tokio::{time, try_join};
+
+#[tokio::main(core_threads = 2)]
+async fn main() -> Result<(), Box<dyn Error + Send + Sync>> {
+    let mut mqttoptions = MqttOptions::new("homie_example", "test.mosquitto.org", 1883);
+    mqttoptions.set_keep_alive(5);
+
+    let (mut homie, mqtt_handle) =
+        HomieDevice::builder("homie/example", "Homie example", mqttoptions)
+            .spawn()
+            .await?;
+
+    let handle: JoinHandle<Result<(), Box<dyn Error + Send + Sync>>> = task::spawn(async move {
+        println!("init");
+        time::delay_for(Duration::from_secs(5)).await;
+        homie.ready().await?;
+        println!("ready");
+        time::delay_for(Duration::from_secs(5)).await;
```

Comment:
```suggestion

        homie.ready().await?;
        println!("ready");

        time::delay_for(Duration::from_secs(5)).await;
```
A bit of whitespace would group the print with the thing that it's reporting on a bit better.

This is only confusing because it's more common to print before doing a thing rather than after. In practice, these prints will probably happen before the state transition event gets serviced anyway, so maybe they should just go before?
```suggestion

        println!("Transitioning to `ready`");
        homie.ready().await?;

        time::delay_for(Duration::from_secs(5)).await;
```

Reply:
Added whitespace, but kept the order.